testing/trino-testing/src/main/java/io/trino/testing/containers/TestContainers.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.testing.containers;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.TestcontainersConfiguration;
+
+import java.io.Closeable;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.Boolean.parseBoolean;
+import static java.lang.System.getenv;
+
+public final class TestContainers
+{
+    // To reuse container set TESTCONTAINERS_REUSE_ENABLE=true environment variable.
+    // TESTCONTAINERS_REUSE_ENABLE is an environment variable defined in testcontainers library.
+    private static final boolean TESTCONTAINERS_REUSE_ENABLE = parseBoolean(getenv("TESTCONTAINERS_REUSE_ENABLE"));
+
+    private TestContainers() {}
+
+    // You should not close the container directly if you want to reuse it.
+    // Instead you should close closeable returned by {@link this::startOrReuse}
+    public static Closeable startOrReuse(GenericContainer<?> container)
+    {
+        boolean reuse = TestcontainersConfiguration.getInstance().environmentSupportsReuse();
+        checkState(reuse == TESTCONTAINERS_REUSE_ENABLE, "Unable to enable or disable container reuse");
```

Comment:
Why we need both settings: the `TESTCONTAINERS_REUSE_ENABLE` env var & `~/.testcontainers.properties` entry? 
wouldn't it be possible to have this controlled with one toggle?

Reply:
It is controlled with one toggle. The env var `TESTCONTAINERS_REUSE_ENABLE`. It is testcontainers that is changing the `~/.testcontainers.properties` when we change `TESTCONTAINERS_REUSE_ENABLE`. Basically we verify if testcontainers behaviour didn't change.

Reply:
are you saying that when i set env TESTCONTAINERS_REUSE_ENABLE=true, TC will automatically change my `~/.testcontainers.properties`?

but then, when i unset TESTCONTAINERS_REUSE_ENABLE, it doesn't revert the file, so i need to manually clean it up

Reply:
Not exactly. 

I am saying that using `TESTCONTAINERS_REUSE_ENABLE` overrides memory representation of `~/.testcontainers.properties`. This is testcontainers implementation detail that we check has not changed when the library is updated.

I think I know what is your problem. You changed `~/.testcontainers.properties` globally (the thing that we were not planning to change when this code was added) and now you need to set `TESTCONTAINERS_REUSE_ENABLE` to make it work. Maybe we can relax this check to skip it when `! TESTCONTAINERS_REUSE_ENABLE`.

Reply:
Oh, so i roll back my changes in `~/.testcontainers.properties` and now the `TESTCONTAINERS_REUSE_ENABLE` env var is all i needed to know about?
thanks, that's exactly what i need.

Reply:
> the TESTCONTAINERS_REUSE_ENABLE env var is all i needed to know

Yes. That was the plan to have a single and simple toggle.